React/game/src/app/components/authRoute/index.js

```diff
@@ -0,0 +1,32 @@
+import React, { PureComponent } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+class ValidateRouter extends PureComponent {
+  renderRoute = props => {
+    const { isPrivate, component: Component, isLogin } = this.props;
+    if (isLogin && !isPrivate) {
```

Comment:
here if the route is not private there's no need to check `isLogin`, since the route is public...
In this conditional, if the route is not private you are redirecting to `/game` which is a  private route by requirements... I'd double check this.

Reply:
the idea is, if the route is not private, redirect it to one with the properties that arrive, redirect to Login using the contrary arguments